openfasoc/generators/glayout/glayout/flow/blocks/strongARM_US/strongARM.py

```diff
@@ -0,0 +1,160 @@
+    mystrongARM=Component(name="mystrongARM")
+    diffp = diffPair(pdk, diffp_w, diffp_l)
+    diffp_ref = prec_ref_center(diffp)
+    mystrongARM.add(diffp_ref)
+    
+
+    cross_couple = cross_coupled_load(pdk, ccinv_col)
+    cross_couple_ref = prec_ref_center(cross_couple)
+    mystrongARM.add(cross_couple_ref)
+    
+
+    clk_nmos = nmos(pdk, width=4, fingers=clk_fing, rmult=1, with_substrate_tap=False, with_dnwell=False)
+    clk_nmos_ref = prec_ref_center(clk_nmos)
+    mystrongARM.add(clk_nmos_ref)
+
+    reset_pmos_right = two_pfet_interdigitized(pdk, numcols=2, dummy=False, with_substrate_tap=False, with_tie=True, width=reset_w, length=reset_l, rmult=1)
+    reset_pmos_ref1 = prec_ref_center(reset_pmos_right)
+    mystrongARM.add(reset_pmos_ref1)
+    offsety_reset_pmos=evaluate_bbox(cross_couple)[1]/4
+
+    reset_pmos_left = two_pfet_interdigitized(pdk, numcols=2, dummy=False, with_substrate_tap=False, with_tie=True, width=reset_w, length=reset_l, rmult=1)
+    reset_pmos_ref2 = prec_ref_center(reset_pmos_left)
+    mystrongARM.add(reset_pmos_ref2)
+    offsetx_cross = (evaluate_bbox(cross_couple)[0]-evaluate_bbox(reset_pmos_left)[0])/2
+
+    offsety_diffp = (evaluate_bbox(diffp)[1]-evaluate_bbox(clk_nmos)[1])/2
+    offsety_cross = (evaluate_bbox(cross_couple)[1]-evaluate_bbox(diffp)[1])/2
+
+    movey(reset_pmos_ref2, evaluate_bbox(clk_nmos_ref)[1]+pdk.util_max_metal_seperation() + offsety_diffp + evaluate_bbox(diffp)[1]+pdk.util_max_metal_seperation() + offsety_cross+offsety_reset_pmos)
+    movex(cross_couple_ref, evaluate_bbox(reset_pmos_left)[0]+pdk.util_max_metal_seperation()+offsetx_cross)
+    movex(diffp_ref, evaluate_bbox(reset_pmos_left)[0]+pdk.util_max_metal_seperation()+offsetx_cross)
+    movex(clk_nmos_ref, evaluate_bbox(reset_pmos_left)[0]+pdk.util_max_metal_seperation()+offsetx_cross)
+    movey(cross_couple_ref, evaluate_bbox(clk_nmos_ref)[1]+pdk.util_max_metal_seperation() + offsety_diffp + evaluate_bbox(diffp)[1]+pdk.util_max_metal_seperation() + offsety_cross)
+    movey(diffp_ref, evaluate_bbox(clk_nmos_ref)[1]+pdk.util_max_metal_seperation()+offsety_diffp)
+    movey(reset_pmos_ref1, evaluate_bbox(clk_nmos_ref)[1]+pdk.util_max_metal_seperation() + offsety_diffp + evaluate_bbox(diffp)[1]+pdk.util_max_metal_seperation() + offsety_cross + offsety_reset_pmos)
+    movex(reset_pmos_ref1, evaluate_bbox(reset_pmos_left)[0]+pdk.util_max_metal_seperation() + evaluate_bbox(cross_couple)[0]+pdk.util_max_metal_seperation())
+    #print("bbox",evaluate_bbox(clk_nmos_ref)[1])
+    mystrongARM.add_ports(diffp_ref.get_ports_list(), prefix="strongARM_")
+    mystrongARM.add_ports(cross_couple_ref.get_ports_list(), prefix="strongARM_")
+    mystrongARM.add_ports(clk_nmos_ref.get_ports_list(), prefix="strongARM_clk_nmos_")
+    mystrongARM.add_ports(reset_pmos_ref1.get_ports_list(), prefix="strongARM_clk_reset_pmos_r_")
+    mystrongARM.add_ports(reset_pmos_ref2.get_ports_list(), prefix="strongARM_clk_reset_pmos_l_")
```

Comment:
add ports after all the routing has been done, it is beneficial to include the routes' ports as well